Provide Branch Hosting for GitHub repo with Netlify (#32)
wmde/new-lexeme-special-page: README.md

1. Clone this repository
2. Install dependecies with `npm i`
3. Test and lint with `npm test`

## Manual Deployment

Before making a deploy make sure you make a build locally of the content that should be deployed by running `npm run buildPreview`.

> Afterwards, run `npm run netlify deploy` to run manual deployments to a unique address that will be generated at the time of deploy. 

Reviewer: It’s asking me for a publish directory. What’s the right option – `dist` or `.`?

Author: dist.

Author: do you think I should add that information to the docs??? i guess that can also be inferred from looking at what buildPreview does, but I can also make it clearer here.

Reviewer: I definitely think it should be added to the docs, yes.

Author: i added it.

Reviewer: Hm, I’m not sure that would answer my question… I meant adding it to the paragraph about `npm run netlify deploy`, since that is the command asking about the publish directory.

Reviewer: Maybe something like:

> Afterwards, run `npm run netlify deploy` and tell it to publish the `dist` directory, to manually deploy the build to a unique address that will be generated at the time of deploy.

Author: I thought this can be inferred from the previous paragraph. But if it's clearer for you adding it as well to this paragraph I can add it.

Reviewer: I’d add it here and remove it from the previous paragraph to avoid duplication :) but I think it helps if it’s closer to the deploy command.

Author: i think it's good to let the users know where the bundle is being built. this is why i thought it's a better idea to put it in the build step.

Author: maybe we can ask a third person what they think? :)

Reviewer 2: I think I see the issue. I wonder if we should go by the old principle of "show, don't tell" here. What if we add a script that actually does all the steps for the developer, and then just instruct them to `npm run deployCommit` or something similar?

I think we can use both `buildPreview` and `netlify deploy --dir="./dist"` for that respect

To deploy to [https://new-lexeme-special-page.netlify.app](https://new-lexeme-special-page.netlify.app) run `npm run netlify deploy --prod`. 

> To be able to run the netlify commands you will need to run `netlify login` once after installing netlify to authenticate and obtain a token.

Reviewer: What does installing netlify mean here? Is it part of the `npm install` or something else? (And if it is part of the `npm install`, shouldn’t the command be `npm run netlify login`, so it uses the `netlify` command from `node_modules`?)

Author: ah yes, it should!!! my bad. i will fix it. installing netlify means running `npm install netlify-cli --save-dev`

Author: Done, the command is updated!

1. Clone this repository
2. Install dependecies with `npm i`
3. Test and lint with `npm test`

## Deploy Local Commits

### The first time you run a deploy

> To be able to run the netlify commands you will need to run `npm run netlify login` once after installing netlify to authenticate and obtain a token.

Reviewer 2: Same here, I think we could go with just `npx`:

```suggestion
To be able to run the netlify commands locally you will need to run `npx netlify login` once after `npm install` to authenticate and obtain a token.
```

### Every time when doing a deploy

> To create a deploy of the current commit, run `npm run deployCommit`. This will create a build in the `./dist` folder and create a manual deployment to a unique address that will be generated at the time of deploy. 

Reviewer 2: Nice and snazzy! Thank you!

> To deploy to [https://new-lexeme-special-page.netlify.app](https://new-lexeme-special-page.netlify.app) run `npx netlify deploy --prod`.

Reviewer: Shouldn’t this command also have `--dir='./dist'`? (Or maybe `npm run deployCommit --prod` would also work?)

Author: also the current command works.

Author: with the current state of the deployCommit script. `npm run deployCommit --prod` doesn't deploy to production.

Author: it deploys by default to `./dist`. i dont understand the need of adding it to three places now.

Reviewer: That’s confusing, does it not ask about the directory in that case? (I haven’t tried it out yet.)

Reviewer: Here’s what I get:

```
$ npx netlify deploy --prod
Please provide a publish directory (e.g. "public" or "dist" or "."):
/var/www/html/wiki1/extensions/WikibaseLexeme/resources/special/new-lexeme
? Publish directory (.) 
```

Reviewer: Maybe it remembers your choice, and that’s why, on your end, it deploys `./dist` by default?

Author: The directory is specified once when you login for the first time. You won't be asked for the directory again.

Reviewer: I do have an account, it made me create one. I don’t think it asked me about this setting at any point.

Reviewer: > with the current code of the deployCommit script. `npm run deployCommit --prod` doesn't deploy to production.

My bad, needs two more dashes. `npm run deployCommit -- --prod` seems to do the right thing (but I get an error, because my personal account isn’t allowed to deploy to this site, of course).

Reviewer 2: I would suggest removing this instruction altogether, since we will most probably not need to deploy to any production site. We only want commit deployments. Our "prod" is WikibaseLexeme itself.

> To deploy to [https://new-lexeme-special-page.netlify.app](https://new-lexeme-special-page.netlify.app) run `npm run netlify deploy --prod`.

Reviewer 2: Since we don't have the `netlify` script anymore I think it would be:

```suggestion
To deploy to [https://new-lexeme-special-page.netlify.app](https://new-lexeme-special-page.netlify.app) run `npx netlify deploy --prod`.
```

Author: true!

1. Clone this repository
2. Install dependecies with `npm i`
3. Test and lint with `npm test`

> ## Manual Deployment

Reviewer 2: This deploys the current commit to a uniquly generated URL, right? Maybe this is a more accurate title:

```suggestion
## Deploy Local Commits
```
